### PR TITLE
add Manhuagui(zh-CN)

### PR DESCRIPTION
**Pull request by feilongfl**
add chinese comic source

### CODE REVIEW

File: src/zh/manhuagui/src/eu/kanade/tachiyomi/extension/zh/manhuagui/Manhuagui.kt [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+import org.jsoup.nodes.Element
+
+
+class Manhuagui : ParsedHttpSource() {
+
+    override val name = "漫画柜"
+    override val baseUrl = "https://www.manhuagui.com"
+    override val lang = "zh"
+    override val supportsLatest = true
+    val imageServer = arrayOf("https://i.hamreus.com")
```
Comment (arkon) [inline src/zh/manhuagui/src/eu/kanade/tachiyomi/extension/zh/manhuagui/Manhuagui.kt]: Does this need to be an array?

File: src/zh/manhuagui/src/eu/kanade/tachiyomi/extension/zh/manhuagui/Manhuagui.kt [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+    private val gson = Gson()
+
+    override fun popularMangaRequest(page: Int) = GET("$baseUrl/list/view_p$page.html", headers)
+    override fun latestUpdatesRequest(page: Int) = GET("$baseUrl/list/update_p$page.html", headers)
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val url = HttpUrl.parse("https://www.manhuagui.com/s/${query}_p$page.html")?.newBuilder()
```
Comment (arkon) [inline src/zh/manhuagui/src/eu/kanade/tachiyomi/extension/zh/manhuagui/Manhuagui.kt]: Use `baseUrl` here.

Is the use of the parser here to encode the `query`?

File: src/zh/manhuagui/src/eu/kanade/tachiyomi/extension/zh/manhuagui/Manhuagui.kt [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    """
+
+    override fun pageListParse(document: Document): List<Page> {
+        val html = document.html()
+        val re = Regex("""window\[".*?"\](\(.*\)\s*\{[\s\S]+\}\s*\(.*\))""")
+        val imgCode = re.find(html)?.groups?.get(1)?.value
+//        val imgCode = """'test'"""
```
Comment (arkon) [inline src/zh/manhuagui/src/eu/kanade/tachiyomi/extension/zh/manhuagui/Manhuagui.kt]: Delete these commented out test/log lines.

File: src/zh/manhuagui/src/eu/kanade/tachiyomi/extension/zh/manhuagui/Manhuagui.kt [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+        return imageJson.files!!.mapIndexed { i, imgStr ->
+            val imgurl = "${imageServer[0]}${imageJson.path}$imgStr?cid=${imageJson.cid}&md5=${imageJson.sl?.md5}"
+            Log.i("image", imgurl)
+            Page(i, "", imgurl)
+        }
+    }
+
+    override fun imageUrlParse(document: Document) = ""
+
+    private class GenreFilter(genres: Array<String>) : Filter.Select<String>("Genre", genres)
```
Comment (arkon) [inline src/zh/manhuagui/src/eu/kanade/tachiyomi/extension/zh/manhuagui/Manhuagui.kt]: You might as well remove this filter if it's not actually used.

File: src/zh/manhuagui/src/eu/kanade/tachiyomi/extension/zh/manhuagui/Manhuagui.kt (hunk 1 above)
Comment (feilongfl) [reply src/zh/manhuagui/src/eu/kanade/tachiyomi/extension/zh/manhuagui/Manhuagui.kt]: I think this website maybe add more server in future.

File: src/zh/manhuagui/src/eu/kanade/tachiyomi/extension/zh/manhuagui/Manhuagui.kt (hunk 2 above)
Comment (feilongfl) [reply src/zh/manhuagui/src/eu/kanade/tachiyomi/extension/zh/manhuagui/Manhuagui.kt]: it's utf-8 uncode,java can auto solve it

File: src/zh/manhuagui/src/eu/kanade/tachiyomi/extension/zh/manhuagui/Manhuagui.kt (hunk 3 above)
Comment (feilongfl) [reply src/zh/manhuagui/src/eu/kanade/tachiyomi/extension/zh/manhuagui/Manhuagui.kt]: I got it

File: src/zh/manhuagui/src/eu/kanade/tachiyomi/extension/zh/manhuagui/Manhuagui.kt (hunk 4 above)
Comment (feilongfl) [reply src/zh/manhuagui/src/eu/kanade/tachiyomi/extension/zh/manhuagui/Manhuagui.kt]: OK,I'll fix and test it

Comment (feilongfl) [reply src/zh/manhuagui/src/eu/kanade/tachiyomi/extension/zh/manhuagui/Manhuagui.kt]: this is `abstract` in parent class,I can't remove it.

**Comment by feilongfl** [comment]
@arkon 
I've removed comment.

File: src/zh/manhuagui/src/eu/kanade/tachiyomi/extension/zh/manhuagui/Manhuagui.kt (hunk 2 above)
Comment (feilongfl) [reply src/zh/manhuagui/src/eu/kanade/tachiyomi/extension/zh/manhuagui/Manhuagui.kt]: I've fixed it to use `baseurl`

Comment (feilongfl) [reply src/zh/manhuagui/src/eu/kanade/tachiyomi/extension/zh/manhuagui/Manhuagui.kt]: I've fix this

**Pull request merged**